scripts/generate-api.ts

```diff
@@ -0,0 +1,18 @@
+import { resolve } from 'node:path'
+import { config } from 'dotenv'
+import { generateApi } from 'swagger-typescript-api'
+
+config({
+	path: [resolve(process.cwd(), '.env'), resolve(process.cwd(), '.env.local')],
+})
+
+const backendUrl = process.env.VITE_BACKEND_URL
+
+if (!backendUrl) {
+	throw new Error('Backend URL not set!')
+}
+
+generateApi({
+	url: `${backendUrl}/swagger/v1/swagger.json`,
+	output: resolve(process.cwd(), 'src'),
+})
```

Comment:
Do we not want this to generate in the same way as the main frontend repo?
https://github.com/leaderboardsgg/leaderboard-site/blob/main/scripts/generate-api.js

Reply:
I didn't think it mattered.

Reply:
Ideally, I'd like to keep it with a similar structure, just so it's easier to swap headspaces between repos.